android/app/src/main/java/com/zulipmobile/notifications/NotificationUiManager.kt

```diff
@@ -323,23 +343,7 @@ private fun updateNotification(
         setContentIntent(
             PendingIntent.getActivity(context, 0,
                 Intent(Intent.ACTION_VIEW, intentUrl, context, MainActivity::class.java)
-                    .setFlags(
-                        // See these sections in the Android docs:
-                        //   https://developer.android.com/guide/components/activities/tasks-and-back-stack#TaskLaunchModes
-                        //   https://developer.android.com/reference/android/content/Intent#FLAG_ACTIVITY_CLEAR_TOP
-                        //
-                        // * From the doc on `PendingIntent.getActivity` at
-                        //     https://developer.android.com/reference/android/app/PendingIntent#getActivity(android.content.Context,%20int,%20android.content.Intent,%20int)
-                        //   > Note that the activity will be started outside of the context of an
-                        //   > existing activity, so you must use the Intent.FLAG_ACTIVITY_NEW_TASK
-                        //   > launch flag in the Intent.
-                        //
-                        // * The flag FLAG_ACTIVITY_CLEAR_TOP is mentioned as being what the
-                        //   notification manager does; so use that.  It has no effect as long
-                        //   as we only have one activity; but if we add more, it will destroy
-                        //   all the activities on top of the target one.
-                        Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP
-                    )
+                    .setFlags(intentFlags)
                     .putExtra(EXTRA_NOTIFICATION_DATA, bundleOf(*fcmMessage.dataForOpen())),
```

Comment:
Here's the line in the existing code where we supply the data that the JS side of the app relies on in order to know where it should navigate to when opening the notification.

Reply:
Depending on the outcome of the below, I take it we would want a `dataForSummaryOpen` that only includes the realm_uri and user_id?

Reply:
Sure, that sounds good.